testing/go/types_test.go

```diff
@@ -20,6 +20,685 @@ import (
+				Expected: []sql.Row{
+					{1, "abcdefghij"},
+					{2, "klmnopqrst"},
+				},
+			},
+		},
+	},
+	{
+		Name: "Cidr type",
+		Skip: true,
+		SetUpScript: []string{
+			"CREATE TABLE t_cidr (id INTEGER primary key, v1 CIDR);",
+			"INSERT INTO t_cidr VALUES (1, '192.168.1.0/24'), (2, '10.0.0.0/8');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM t_cidr ORDER BY id;",
+				Expected: []sql.Row{
+					{1, "192.168.1.0/24"},
+					{2, "10.0.0.0/8"},
+				},
+			},
+		},
+	},
+	{
+		Name: "Circle type",
+		Skip: true,
+		SetUpScript: []string{
+			"CREATE TABLE t_circle (id INTEGER primary key, v1 CIRCLE);",
+			"INSERT INTO t_circle VALUES (1, '<(1,2),3>'), (2, '<(4,5),6>');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				// TODO: might need a GMS type here, not a string
+				Query: "SELECT * FROM t_circle ORDER BY id;",
+				Expected: []sql.Row{
+					{1, "<(1,2),3>"},
+					{2, "<(4,5),6>"},
+				},
+			},
+		},
+	},
+	{
+		Name: "Date type",
+		SetUpScript: []string{
+			"CREATE TABLE t_date (id INTEGER primary key, v1 DATE);",
+			"INSERT INTO t_date VALUES (1, '2023-01-01'), (2, '2023-02-02');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Skip:  true, // TODO: these are coming back as date time objects, not dates
+				Query: "SELECT * FROM t_date ORDER BY id;",
+				Expected: []sql.Row{
+					{1, "2023-01-01"},
+					{2, "2023-02-02"},
+				},
+			},
+		},
+	},
+	{
+		Name: "Double precision type",
+		SetUpScript: []string{
+			"CREATE TABLE t_double_precision (id INTEGER primary key, v1 DOUBLE PRECISION);",
+			"INSERT INTO t_double_precision VALUES (1, 123.456), (2, 789.012);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM t_double_precision ORDER BY id;",
+				Expected: []sql.Row{
+					{1, 123.456},
+					{2, 789.012},
+				},
+			},
+		},
+	},
+	{
+		Name: "Inet type",
+		Skip: true,
+		SetUpScript: []string{
+			"CREATE TABLE t_inet (id INTEGER primary key, v1 INET);",
+			"INSERT INTO t_inet VALUES (1, '192.168.1.1'), (2, '10.0.0.1');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM t_inet ORDER BY id;",
+				Expected: []sql.Row{
+					{1, "192.168.1.1"},
+					{2, "10.0.0.1"},
+				},
+			},
+		},
+	},
+	{
+		Name: "Integer type",
+		SetUpScript: []string{
+			"CREATE TABLE t_integer (id INTEGER primary key, v1 INTEGER);",
+			"INSERT INTO t_integer VALUES (1, 123), (2, 456);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM t_integer ORDER BY id;",
+				Expected: []sql.Row{
+					{1, 123},
+					{2, 456},
+				},
+			},
+		},
+	},
+	{
+		Name: "Interval type",
+		Skip: true,
+		SetUpScript: []string{
+			"CREATE TABLE t_interval (id INTEGER primary key, v1 INTERVAL);",
+			"INSERT INTO t_interval VALUES (1, '1 day 3 hours'), (2, '2 hours 30 minutes');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				// TODO: might need a GMS type here, not a string (psql output is different than below)
```

Comment:
I'd look at the wire return here. PSQL is free to reinterpret results, which may be specific to PSQL. It's possible that we're supposed to return an integer in like microseconds, and PSQL just reinterprets it to some other displayable format. This goes for the other types as well where you've written something similar.

Reply:
Yeah, I'm thinking there will probably be a lot of this